packages/core/src/2d/assembler/SimpleSpriteAssembler.ts

```diff
@@ -74,15 +82,25 @@ export class SimpleSpriteAssembler {
     vertices[offset + 28] = top;
   }
 
-  static updateColor(renderer: SpriteRenderer): void {
+  static updateColor(renderer: SpriteRenderer, alpha: number = 1): void {
     const subChunk = renderer._subChunk;
     const { r, g, b, a } = renderer.color;
+    const finalAlpha = a * alpha;
```

Comment:
Above three lines and following line can be optimized using destructuring to reduce redundancy and improve readability. Here’s an optimized version:

const { _subChunk: { chunk: { vertices } }, color: { r, g, b, a } } = renderer;
const finalAlpha = a * alpha;